### PR TITLE
scaleup cluster

### PR DESCRIPTION
**Pull request by Ameenasuhani**
# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs

# Checklist:

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [x] Review the automation design
- [x] Implement the test script and perform test runs
- [x] Submit PR for code review and approve
- [x] Update Polarin Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test

Logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1603104569414/result.html

### CODE REVIEW

**Comment by sunilangadi2** [comment]
@Ameenasuhani Please cover grafana daemon as well in this scale up suite.

File: suites/nautilus/ansible/sanity_ceph_ansible_scale.yaml [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+                    osd_pool_default_pgp_num: 64
+                    mon_max_pg_per_osd: 1024
+                  mon:
+                    mon_allow_pool_delete: true
+              cephfs_pools:
+                - name: "cephfs_data"
+                  pgs: "8"
+                - name: "cephfs_metadata"
+                  pgs: "8"
+          add:
```
Comment (sunilkumarn417) [inline suites/nautilus/ansible/sanity_ceph_ansible_scale.yaml:78]: does scale up works for multiple nodes?

Comment (sunilkumarn417) [reply suites/nautilus/ansible/sanity_ceph_ansible_scale.yaml:78]: Example: running a playbook to scale more than one rgw(s).

File: suites/nautilus/ansible/sanity_ceph_ansible_scale.yaml [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+                  node-name: .*node5.*
+                  demon:
+                      - mds
+      desc: add mds daemon
+
+   - test:
+      name: config roll over client daemon
+      polarion-id: CEPH-9581
+      module: test_ansible_roll_over.py
+      config:
```
Comment (sunilkumarn417) [inline suites/nautilus/ansible/sanity_ceph_ansible_scale.yaml:128]: Is ansible config(ansi_config) parameters are common across all playbook and considered at every execution?. if not used, can we eliminate it?

File: suites/nautilus/ansible/sanity_ceph_ansible_scale.yaml [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+                    osd_pool_default_pgp_num: 64
+                    mon_max_pg_per_osd: 1024
+                  mon:
+                    mon_allow_pool_delete: true
+              cephfs_pools:
+                - name: "cephfs_data"
+                  pgs: "8"
+                - name: "cephfs_metadata"
+                  pgs: "8"
+          add:
```
Comment (sunilkumarn417) [inline suites/nautilus/ansible/sanity_ceph_ansible_scale.yaml:195]: does scale-up also updates inventory file?

File: suites/nautilus/ansible/sanity_ceph_ansible_scale.yaml [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+      config:
+          is_mixed_lvm_configs: True
+          ansi_config:
+              ceph_test: True
+              ceph_origin: distro
+              ceph_stable_release: nautilus
+              ceph_repository: rhcs
+              osd_scenario: lvm
+              osd_auto_discovery: False
+              journal_size: 1024
```
Comment (sunilkumarn417) [inline suites/nautilus/ansible/sanity_ceph_ansible_scale.yaml:176]: is journal size needed for nautilus ceph cluster?

**Review by sunilkumarn417 — CHANGES_REQUESTED** [review]
- Does this scale up scenarios should be covered for luminous also?
- Please squash the  commits if needed.

File: suites/nautilus/ansible/sanity_ceph_ansible_scale.yaml (hunk 1 above)
Comment (Ameenasuhani) [reply suites/nautilus/ansible/sanity_ceph_ansible_scale.yaml:78]: Ceph-ansible definitely supports adding multiple nodes at once using playbook but the cephci framework doesn't hence I have raised an issue earlier https://github.com/red-hat-storage/cephci/issues/224

File: suites/nautilus/ansible/sanity_ceph_ansible_scale.yaml (hunk 3 above)
Comment (Ameenasuhani) [reply suites/nautilus/ansible/sanity_ceph_ansible_scale.yaml:195]: yes it does

File: suites/nautilus/ansible/sanity_ceph_ansible_scale.yaml (hunk 2 above)
Comment (Ameenasuhani) [reply suites/nautilus/ansible/sanity_ceph_ansible_scale.yaml:116]: https://github.com/red-hat-storage/cephci/blob/master/tests/ceph_ansible/test_ansible_roll_over.py#L66 yes they are common and considered for every execution

**Comment by Ameenasuhani** [comment]
@sunilangadi2 sure will add support for dashboard

**Comment by Ameenasuhani** [comment]
logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1604311580369/result.html

**Comment by Ameenasuhani** [comment]
logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1605635411071/result.html

File: suites/nautilus/ansible/sanity_ceph_ansible_scale.yaml (hunk 4 above)
Comment (Ameenasuhani) [reply suites/nautilus/ansible/sanity_ceph_ansible_scale.yaml]: I have made the changes

**Comment by Ameenasuhani** [comment]
> * Does this scale up scenarios should be covered for luminous also?
> * Please squash the  commits if needed.

yes we need to

> * Does this scale up scenarios should be covered for luminous also?
> * Please squash the  commits if needed.

yes this is required in luminous too(rgw, mds, nfs )

File: suites/nautilus/ansible/sanity_ceph_ansible_scale.yaml (hunk 2 above)
Comment (sunilkumarn417) [reply suites/nautilus/ansible/sanity_ceph_ansible_scale.yaml:140]: Do we need ansible_config even after deployment? other than only few parameters which is required for upgrade or to run specific playbook.

**Comment by Ameenasuhani** [comment]
changes requested can be tracked here https://github.com/red-hat-storage/cephci/pull/269

**Pull request closed without merging**